kixi-io/Ki.KD-TypeScript
# nz - Setup Nx workspace

This PR converts the repo into an Nx workspace with the following features:

- Linting via eslint
- Formatting via prettier
- Testing via jest
- See README for cli commands

For further discussion:
- Package name - currently this is set to `@kixi-io/ki.kd-typescript`
- Package distribution currently has a tslib dependency. This can be squashed into the build which will increase the build size. See `--importHelpers` in Typescript docs for details.
- Lots of linting issues currently in the code base. May want to evaluate linting rules that should be enforced and which should be relaxed.

Other Notes:
- There was a circular dependency with the KD > KDInterp > listOf > KD chain. This was resolved using an internal export file. See `_internal.ts` for details.
- QA class was removed in favor of jest. Tests were re-written using jest syntax.

Suggested follow up:
- Documentation
- More tests
    - Statement coverage: 82%
    - Branch coverage: 62%
    - Function coverage: 65%
    - Line coverage: 84%
- Setup publishing to npm
- Setup build pipeline using Github Actions
- Add commit hooks to prevent unformatted code from being committed
